relion/protocols/protocol_base.py

```diff
@@ -1067,8 +1084,13 @@ def _getScratchDir(self):
         """ Returns the scratch dir value without spaces.
          If none, the empty string will be returned.
         """
-        scratchDir = self.scratchDir.get() or ''
-        return scratchDir.strip()
+        scratchDir = ''
+        if self.useScratch == USE_SCIPION_SCRATCH:
+            scratchDir = self._getTmpPath()
```

Comment:
The point of this PR is that tmp is a link to a unique dir inside SCIPION_SCRATCH when defined. I added a validator to prevent using this option when SCIPION_SCRATCH is not defined.

Reply:
If I were using SCIPION_SCRATCH alone, all concurrent relion protocols would try to use SCIPION_SCRATCH/relion_volatile simultaneously, leading to collisions (they would overwrite delete each other's data)

Reply:
Using the tmp directory from the protocol (only allowed if SCIPION_SCRATCH is defined) would place each protocol's scratch in SCIPION_SCRATCH/<sufficiently unique dir>/relion_volatile

Reply:
Oh I see, I forgot about the unique folder inside: https://github.com/scipion-em/scipion-pyworkflow/blob/devel/pyworkflow/utils/path.py#L189 You are right!